### DISCUSSION TITLE
Removing cells from analysis

### DISCUSSION THREAD

**Discussion by NoemieL** [GreenleafLab/ArchR#732] [Questions / Documentation]
Hi,
I am analyzing magnetic-column sorted cells by scATACseq and scRNAseq. After integrating scRNAseq data, I identified contaminated cells (cells of non interest) that I want to removed from my analysis. 
    1) I was using subsetCells to remove these cells but I read different comments and I was wandering if I need to use subsetArchRProject instead? I don't understand what are really the consequences linked to the choice between subsetArchRProject and subsetCells (except that in one case we removed cells from the Arrow file) on the rest of the analysis since I am saving it as a new project to work on (proj_all_8=subsetCells(ArchRProj = proj_all_3, cellNames = kept_cells).
   2) After removing the cells, do we need to remove the matched cells on the scRNAseq data and reperform the scRNAseq integration step? I was only reperforming the clustering step, but it is a question I had.
   3) Since I am using the scATACseq and scRNAseq data from the same patients samples, I was wandering if there is a way to performed only the constrained integration based on samples and not based on clusters? I have sometimes ATACseq cells from one patient that match with RNAseq cells from other patients if I used the clusters as shown in the manual.

Thanks in advances for your help

**Comment by rcorces** [comment]
1) I would always use subsetArchRProject() which is designed to handle all aspects of the project. subsetCells(() actually only changes the data stored in cellColData.
2) Theoretically, this should happen automatically if you use subsetArchRProject() on and object with integrated RNA.
3) You can constrain the integration however you want. See the function documentation.
